GameOfRunes/GameOfRunes/GameEngine/Components/MultiplierComponent.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  MultiplierComponent.swift
+//  GameOfRunes
+//
+//  Created by Andy on 25/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import SpriteKit
+import GameplayKit
+
+class MultiplierComponent: GKComponent, Component {
+    var type: ComponentType {
+        .multiplierComponent
+    }
+    var multiplier: Int = 1
```

Comment:
instead of doing this right, add a multiplier to `metadata`, then for this component it will just be an observable on `metadata`'s multiplier, then add this component to both `ComboEntity` and `PlayerEntity`, then when increasing score for `PlayerEntity` search for the `MultiplierComponent` and use it to update the score instead of increasing it when the combo expires. When the combo expires, reset the multiplier to 1.